packages/block-editor/src/store/actions.js

```diff
@@ -1210,34 +1080,29 @@ export function __unstableMarkNextChangeAsNotPersistent() {
 }
 
 /**
- * Returns an action object used in signalling that the last block change is
- * an automatic change, meaning it was not performed by the user, and can be
- * undone using the `Escape` and `Backspace` keys. This action must be called
- * after the change was made, and any actions that are a consequence of it, so
- * it is recommended to be called at the next idle period to ensure all
+ * Action that marks the last block change as an automatic change, meaning it was not
+ * performed by the user, and can be undone using the `Escape` and `Backspace` keys.
+ * This action must be called after the change was made, and any actions that are a
+ * consequence of it, so it is recommended to be called at the next idle period to ensure all
  * selection changes have been recorded.
  */
-export function* __unstableMarkAutomaticChange() {
-	yield { type: 'MARK_AUTOMATIC_CHANGE' };
-	yield __unstableMarkAutomaticChangeFinalControl();
-}
-
-export function __unstableMarkAutomaticChangeFinal() {
-	return {
-		type: 'MARK_AUTOMATIC_CHANGE_FINAL',
-	};
-}
+export const __unstableMarkAutomaticChange = () => ( { dispatch } ) => {
+	dispatch( { type: 'MARK_AUTOMATIC_CHANGE' } );
+	const { requestIdleCallback = ( cb ) => setTimeout( cb, 100 ) } = window;
```

Comment:
I think this would be more readable – feel free to accept or reject as you see fit:

```suggestion
	const requestIdleCallback = window.requestIdleCallback || ( cb ) => setTimeout( cb, 100 );
```

Reply:
I also experimented with several options here and I concluded they are all equally ugly 🙂 So I ended up with the original syntax, unmodified.